src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking_FlowAnalysis.cs

```diff
@@ -890,6 +893,77 @@ static void F3(bool b1, bool b2, object v3)
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "w3").WithLocation(67, 13));
         }
 
+        // PROTOTYPE(NullableReferenceTypes): Review /*T:...*/ and diagnostics.
+        [Fact]
+        public void ConditionalOperator_14()
+        {
+            var source =
+@"interface I<T> { T P { get; } }
+interface IIn<in T> { }
+interface IOut<out T> { T P { get; } }
+class C
+{
+    static void F1(bool b, ref string? x1, ref string y1)
+    {
+        (b ? ref x1 : ref x1)/*T:string?*/.ToString();
+        (b ? ref x1 : ref y1)/*T:string?*/.ToString();
+        (b ? ref y1 : ref x1)/*T:string?*/.ToString();
+        (b ? ref y1 : ref y1)/*T:string!*/.ToString();
+    }
+    static void F2(bool b, ref I<string?> x2, ref I<string> y2)
+    {
+        (b ? ref x2 : ref x2)/*T:I<string?>!*/.P.ToString();
+        (b ? ref y2 : ref x2)/*T:I<string>!*/.P.ToString();
+        (b ? ref x2 : ref y2)/*T:I<string>!*/.P.ToString();
+        (b ? ref y2 : ref y2)/*T:I<string!>!*/.P.ToString();
+    }
+    static void F3(bool b, ref IIn<string?> x3, ref IIn<string> y3)
+    {
+        (b ? ref x3 : ref x3)/*T:IIn<string?>!*/.ToString();
+        (b ? ref y3 : ref x3)/*T:IIn<string!>!*/.ToString();
+        (b ? ref x3 : ref y3)/*T:IIn<string!>!*/.ToString();
+        (b ? ref y3 : ref y3)/*T:IIn<string!>!*/.ToString();
+    }
```

Comment:
This test would perhaps be more useful with a property to access like the previous and next methods inside this test?

Reply:
The property could not be of type `T` though since `T` is contravariant.

---
In reply to: [190949913](https://github.com/dotnet/roslyn/pull/26356#discussion_r190949913) [](ancestors = 190949913)